rusk-wallet/src/bin/main.rs

```diff
@@ -85,10 +85,31 @@ where
 
     // check for connection errors
     match con {
-        Err(Error::RocksDB(e)) => panic!{"Please reset the cache! {e}"},
-        Err(e) => warn!("[OFFLINE MODE]: Unable to connect to Rusk, limited functionality available: {e}"),
+        Err(Error::RocksDB(ref e)) => {
+            match prompt::ask_confirm_erase_cache() {
+                Ok(true) => {
+                    wallet.close();
+
+                    if let Some(io_err) = wallet.delete_cache().err() {
+                        error!("Error while deleting the cache: {io_err}");
+                    }
+
+                    info!("Restart the wallet to create new cache and sync with network");
+                },
+                _ => {
+                    wallet.close();
+
+                    error!("Wallet closed with database error: {e}");
+                }
```

Comment:
matching like this means that you are loosing error information:
The second arm of the match is reached if the user entered `no` for a cache reset but also if something else went wrong and `ask_confirm_erase_cache` returned an `Err`.

I would first get to the value of the prompt and then only match on the response being true of false.

Reply:
If the user answers anything other than yes then we cannot proceed because of db error we need to show the error then to the user, we don't really care about other scenarios

Reply:
code becomes much harder to debug when there cases are grouped like this. So I would argue that we do care